config/config.example.yml

```diff
@@ -42,26 +48,151 @@ services:
+        # config below, if we have an input resource path `/nginx/workspaces/user_xyz/dir1/dir2/file_abc` that matches with
+        # the `nginx_workspace` resource key, the `user` variable name would be matched with `user_xyz` and `synched_file`, with `file_abc`.
+        # Also, this key would need to sync permissions with the `thredds_workspace` resource key, considering the
+        # `permission_mappings` defined below. The `thredds_workspace` would be deduced to the resource path
+        # `/catalog/workspaces/user_xyz/dir1/dir2/subdir/file_abc`.
+        # The types of each segment of this target resource path  would be deduced from the `thredds_workspace` config below.
+        thredds_workspace:
+          - name: catalog
+            type: service
+          - name: workspaces
+            type: directory
+          - name: "{user}"
+            type: directory
+          - name: "**"
+            type: directory
+          - name: subdir
+            type: directory
+          - name: "{synched_file}"
+            type: file
+      geoserver:
+        geoserver_workspace:
+          - name: geoserver
+            type: service
+          - name: workspaces
+            type: workspace
+          - name: "{user}"
+            type: workspace
+          - name: "**"
+            type: workspace
+          - name: "{synched_file}"
+            type: workspace
     # [Required] This section defines an array of permissions mapping between services
-    # In this example, for every Geoserver's `read` permission, a `read` and `browse` permissions will be created in
-    # Thredds. If a `browse` permission is created in Thredds a `read` permission will be added to `Geoserver`.
+    # Each item found in the `permissions_mapping` use the following format :
+    #     resource_key1 : permission(s) (-> or <- or <->) resource_key2 : permission(s)
+    # The permissions can either be a single permission or a list of permissions. Each of those permissions can either
+    # use an implicit format (`<name>` or `<name>-match`) or an explicit format (`<name>-<access>-<scope>`).
+    # When using an implicit format, if the access and/or scope are not specified, it will use the default
+    # access `allow` and/or the default scope `recursive`.
+    # For more info on Magpie permissions :
+    #     https://pavics-magpie.readthedocs.io/en/latest/permissions.html#permission-definition-and-modifiers
+    #     https://pavics-magpie.readthedocs.io/en/latest/permissions.html#permissions-representation
+    # The arrows between the 2 resources indicate the direction of the synchronization, and which resources can be a
+    # `source` or `target` resource.
+    # In the case of `<->` arrow, the synchronization of permissions can be done in either direction. Also, it is
+    # important to note that in thise case, both mapped resources should be matching `variable` names if any, and if one
```

Comment:
Yes, exactly. In the bidirectional case, both need to match all variables of the other resource, since it has to work in both directions.
I added a more explicit sentence : 'each resource needs to match all the variables of the other mapped resource.'